torch_geometric/nn/models/captum.py

```diff
@@ -176,51 +222,54 @@ def to_captum_input(x: Union[Tensor, Dict[EdgeType, Tensor]],
-            than :obj:`x_dict` and :obj:`edge_index_dict`.
+        n_samples (int): The number of samples to explain. (default: :obj:`1`)
+        additional_forward_args: Additional forward arguments of the model
+            being explained which will be added to the generated
+            :obj:`additional_forward_args`. For :class:`Data` this is
+            arguments other than :obj:`x` and :obj:`edge_index`. For
+            :class:`HeteroData` this is arguments other than :obj:`x_dict` and
+            :obj:`edge_index_dict`.
     """
     _raise_on_invalid_mask_type(mask_type)
+    additional_forward_args = additional_forward_args[:]
+    additional_forward_args.insert(0, edge_index)
 
-    additional_forward_args = []
     if isinstance(x, Tensor) and isinstance(edge_index, Tensor):
-        if mask_type == "node":
-            inputs = [x.unsqueeze(0)]
-        elif mask_type == "edge":
-            inputs = [_to_edge_mask(edge_index).unsqueeze(0)]
-            additional_forward_args.append(x)
-        else:
-            inputs = [x.unsqueeze(0), _to_edge_mask(edge_index).unsqueeze(0)]
-        additional_forward_args.append(edge_index)
+        inputs = []
+        if "node" in mask_type:
+            inputs.append(_expand_input(x, n_samples))
+        if "edge" in mask_type:
+            inputs.append(_expand_input(_to_edge_mask(edge_index), n_samples))
+        if mask_type == "edge":
+            additional_forward_args.insert(0, x)
 
     elif isinstance(x, Dict) and isinstance(edge_index, Dict):
         node_types = x.keys()
         edge_types = edge_index.keys()
         inputs = []
         if mask_type == "node":
             for key in node_types:
-                inputs.append(x[key].unsqueeze(0))
+                inputs.append(_expand_input(x[key], n_samples))
         elif mask_type == "edge":
             for key in edge_types:
-                inputs.append(_to_edge_mask(edge_index[key]).unsqueeze(0))
-            additional_forward_args.append(x)
+                inputs.append(
+                    _expand_input(_to_edge_mask(edge_index[key]), n_samples))
+            additional_forward_args.insert(0, x)
         else:
             for key in node_types:
-                inputs.append(x[key].unsqueeze(0))
+                inputs.append(_expand_input(x[key], n_samples))
             for key in edge_types:
-                inputs.append(_to_edge_mask(edge_index[key]).unsqueeze(0))
-        additional_forward_args.append(edge_index)
+                inputs.append(
+                    _expand_input(_to_edge_mask(edge_index[key]), n_samples))
 
     else:
         raise ValueError(
             "'x' and 'edge_index' need to be either"
             f"'Dict' or 'Tensor' got({type(x)}, {type(edge_index)})")
-    additional_forward_args.extend(args)
     return tuple(inputs), tuple(additional_forward_args)
 
 
+# TODO (jinu/ramona): Rethink batch hetero output dicts
 def captum_output_to_dicts(
```

Comment:
@wsad1 What do you think about having a general `captum_output_transform` method for both hetero and nonhetero graphs?

Reply:
Yeah thats a good idea.